### DISCUSSION TITLE
Is there a high-level chat option?

### DISCUSSION THREAD

**Discussion by AnOpenSauceDev** [abetlen/llama-cpp-python#424] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I'm currently working on a project that involves llama-cpp-python's high-level API, and i wanted to know if there was a high-level chat feature, as chat completion tends to produce an unintelligible mess when i try talking to it.

**Describe the solution you'd like**
It would be nice if i could get some documentation on how to properly chat with the ai, or some alternate way to do this, without making my codebase too messy.

**Describe alternatives you've considered**
- Running the model low-level (i couldn't find anything related to -ngl, which made me give up pretty quickly)
- running the chat completion model with a prompt, but that lead to it repeatedly affirming me that Moscow was the largest European city. Either that, or hallucinating quite a lot. (it also couldn't recall what i said either.)

**Additional context**
Model: vicuna 7B
GPU: RTX 3060ti, compiled with cuBLAS
I'm running this native on my Pop_OS machine.

**Comment by gjmulder** [comment, marked as answer]
Check out [oobabooga/text-generation-webui](https://github.com/oobabooga/text-generation-webui) which can use `llama-cpp-python`. I've had a lot of success with [llama-cpp-telegram_bot](https://github.com/innightwolfsleep/llama-cpp-telegram_bot) using [Vicuna and WizardLM GGML models @TheBloke has converted.](https://huggingface.co/TheBloke)

The key is to understand how the models you are using were trained and to ensure your prompt template conforms to the training template.

**Reply by AnOpenSauceDev** [reply]
I tried a few of the tricks they used, and got some good results. Thanks!
